packages/server/src/errors.ts

```diff
@@ -195,90 +56,89 @@ export function fromGraphQLError(error: GraphQLError, options?: ErrorOptions) {
+    this.name = 'ValidationError';
   }
 }
 
-export class AuthenticationError extends ApolloError {
+export class AuthenticationError extends GraphQLError {
   constructor(message: string, extensions?: Record<string, any>) {
-    super(message, 'UNAUTHENTICATED', extensions);
+    super(message, { extensions: { ...extensions, code: 'UNAUTHENTICATED' } });
 
-    Object.defineProperty(this, 'name', { value: 'AuthenticationError' });
+    this.name = 'AuthenticationError';
   }
 }
 
-export class ForbiddenError extends ApolloError {
+export class ForbiddenError extends GraphQLError {
   constructor(message: string, extensions?: Record<string, any>) {
-    super(message, 'FORBIDDEN', extensions);
+    super(message, { extensions: { ...extensions, code: 'FORBIDDEN' } });
 
-    Object.defineProperty(this, 'name', { value: 'ForbiddenError' });
+    this.name = 'ForbiddenError';
   }
 }
 
-export class PersistedQueryNotFoundError extends ApolloError {
+export class PersistedQueryNotFoundError extends GraphQLError {
   constructor() {
-    super('PersistedQueryNotFound', 'PERSISTED_QUERY_NOT_FOUND');
-
-    Object.defineProperty(this, 'name', {
-      value: 'PersistedQueryNotFoundError',
+    super('PersistedQueryNotFound', {
+      extensions: { code: 'PERSISTED_QUERY_NOT_FOUND' },
     });
+
+    this.name = 'PersistedQueryNotFoundError';
   }
 }
 
-export class PersistedQueryNotSupportedError extends ApolloError {
+export class PersistedQueryNotSupportedError extends GraphQLError {
   constructor() {
-    super('PersistedQueryNotSupported', 'PERSISTED_QUERY_NOT_SUPPORTED');
-
-    Object.defineProperty(this, 'name', {
-      value: 'PersistedQueryNotSupportedError',
+    super('PersistedQueryNotSupported', {
+      extensions: { code: 'PERSISTED_QUERY_NOT_SUPPORTED' },
     });
+
+    this.name = 'PersistedQueryNotSupportedError';
   }
 }
 
-export class UserInputError extends ApolloError {
+export class UserInputError extends GraphQLError {
   constructor(message: string, extensions?: Record<string, any>) {
-    super(message, 'BAD_USER_INPUT', extensions);
+    super(message, { extensions: { ...extensions, code: 'BAD_USER_INPUT' } });
 
-    Object.defineProperty(this, 'name', { value: 'UserInputError' });
+    this.name = 'UserInputError';
   }
 }
 
 // TODO(AS4): We added this in AS4. Is that a good idea? We should at least
 // document it, and maybe consider using it for more of the errors in
 // runHttpQuery instead of just returning text/plain errors.
-export class BadRequestError extends ApolloError {
+export class BadRequestError extends GraphQLError {
   constructor(message: string, extensions?: Record<string, any>) {
-    super(message, 'BAD_REQUEST', extensions);
+    super(message, { extensions: { ...extensions, code: 'BAD_REQUEST' } });
 
-    Object.defineProperty(this, 'name', { value: 'BadRequestError' });
+    this.name = 'BadRequestError';
   }
 }
 
 // This function should not throw.
 export function formatApolloErrors(
```

Comment:
Also I think you said this but probably the function should get a better name since there are no more Apollo Errors. normalizeAndFormatErrors?

Reply:
I like "enrich" more than "normalize" since we only add stuff and don't remove stuff.
`formatError` can remove stuff but it's more of "formating" part of this function.
`enrichAndFormatErrors`?

Reply:
The normalize part to me is how we convert non-Error and non-GraphQLError to GraphQLError... but I see what you mean.